src/wasm/wasm-s-parser.cpp

```diff
@@ -3451,7 +3451,7 @@ void SExpressionWasmBuilder::validateHeapTypeUsingChild(Expression* child,
     return;
   }
   if ((!child->type.isRef() && !child->type.isRtt()) ||
-      child->type.getHeapType() != heapType) {
+      !HeapType::isSubType(child->type.getHeapType(), heapType)) {
```

Comment:
Could this function be deduplicated to parsing.h? The `Element` argument could become an optional pointer.

Reply:
It almost can, but the mechanisms they use to throw are also different, which is why this wasn't deduplicated before.

A larger refactoring to unify how they throw might make sense I think.

Reply:
ahhh, I missed that difference. Refactoring at some point makes sense to me, but definitely not urgent 👍